numpy/core/src/multiarray/typeinfo.h

```diff
@@ -1,17 +1,19 @@
 #ifndef _NPY_PRIVATE_TYPEINFO_H_
 #define _NPY_PRIVATE_TYPEINFO_H_
 
-void typeinfo_init_structsequences(void);
+#define PY_SSIZE_T_CLEAN
```

Comment:
Hmm, well, I think everything defines this macro now, but this still seems a strange place for it, could cause redefinition warnings. There is only one other include file that defines it, and only a couple that import `Python.h` (without it). We might want to rationalize all those imports at some point.

Reply:
Should we be defining it via distutils instead?

At any rate, I'd argue that warnings only matter if they cause ci to fail (conversely: if you care about a warning, configure ci to make it fail)

Reply:
I don't see any warnings for this, just noting the potential. What I could see doing is another include file, `npy_python.h` or some such, that pairs them up. The include guard would prevent multiple declarations. Or we can alway fix with a `#if !defined(...)`. Worth noting, but not worrying about yet.

Reply:
> Should we be defining it via distutils instead?

Passing as `-D` to the compiler? Worth thinking about.

Reply:
All good suggestions, thanks for thinking about this,